src/unpage/plugins/aptible/client.py

```diff
@@ -143,7 +144,21 @@ async def item_generator() -> AsyncGenerator[dict[str, Any], None]:
                     yield item
 
             # Fetch all subsequent pages in parallel
-            pages = (self.get(f"{url}?page={p}") for p in range(1, total_pages + 1))
+            parsed_url = urlparse(url)
+            pages = (
+                self.get(
+                    parsed_url._replace(
+                        query=urlencode(
+                            {
+                                **parse_qs(parsed_url.query),
+                                "page": p,
+                            },
+                            doseq=True,
+                        )
+                    ).geturl()
```

Comment:
All of this is just setting `&page={p}` and properly handling cases where query parameters are already provided, etc